### PR TITLE
Initialize restructure of routes directory

### LINKED ISSUE #89: Remove events from data model
Should we still retain the `Event` class in our data model? Considering that our events don't overlap, it might be better to just use separate databases for each event, i.e. just change the mongo URL on the backend for each event.

This would also speed up queries because it would minimize the number of documents each search would have to go over.

### LINKED ISSUE #127: [Feature] Use zod for input validation
### Description

Use [zod](https://zod.dev/) to handle input validation on request handlers

### Designs

_No response_

### PR DESCRIPTION
**Pull request by gramliu**
# Description

* Starts the restructuring of the routes directory for readability
  - Move route implementations from controllers to subdirectories under routes. Each route implementation is in a separate file
  - Move route-level documentation to the route implementation file. Keep route module documentation in subdirectory's `index.ts` file
  - Add zod to route handlers for cleaner and more reliably input validation
  - Replace utility response functions `bad(res, <msg>)`, `notFound()`, and `error` with proper error classes: `BadRequestError`, `NotFoundError`, and `ServerError` which all extend the parent `APIError` class. These are all caught with the top-level synchronous error handler and produces a more readable error and JSON-safe response

* Translated the `POST /auth/register`, `GET /settings` and `PUT /settings` endpoint
* Also fixed the long-standing issue with path resolution
  - In the past, we had to ensure that no path aliases were used in imports (e.g. `import "src/Module"`) and all imports were made using relative imports (e.g. `import ../Module`). This was due to a weird issue with how TS gets compiled to native JS without translating the path aliases. If any path aliases were included, the build would succeed but it would crash in production on startup. This PR resolves this by running production builds with `ts-node --transpileOnly` instead of `tsc && node .`. This allows us to run with path aliases and still maintaining low memory overhead.
* Temporarily disabled tests on CI until we can at least have a single route module migrated  

Closes #127
Closes #89

### CODE REVIEW

**Review by pm3512 — COMMENTED** [review]
Good job!

File: src/errors/ServerError.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import APIError from "./APIError";
+
+/**
+ * Internal server error (404)
```
[the remark this answers is not in the document]
Comment (gramliu) [reply src/errors/ServerError.ts:4]: oops forgot to change this

File: src/routes/settings/getSettings.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ *       404:
+ *          description: User does not exist.
+ *       500:
+ *          description: Internal Server Error.
+ */
+
+/**
+ * Retrieve the application settings
+ */
+export async function getSettings(req: Request, res: Response): Promise<void> {
```
[the remark this answers is not in the document]
Comment (gramliu) [reply src/routes/settings/getSettings.ts:29]: this is the route handler so it handles logic for the REST endpoint. After the migration is complete, the controllers should only be in charge of generic logic. Great point though. I should change the logic of this endpoint to read from the `SettingsController` code just so it's all in one place

File: src/db.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+export const prisma =
+  global.prisma ||
+  new PrismaClient({
+    log:
+      process.env.NODE_ENV === "dev" ? ["query", "error", "warn"] : ["error"],
+  });
+
+if (process.env.NODE_ENV !== "prod") {
+  global.prisma = prisma;
```
[the remark this answers is not in the document]
Comment (gramliu) [reply src/db.ts:16]: Good point, I saw this and completely forgot about it. This is boilerplate code from the [create-t3-app](https://github.com/t3-oss/create-t3-app) template. I honestly don't think there's any reason to not add it to `global` in prod so we can probably just remove the if condition

**Pull request merged**